source/Octopus.Client/Model/OctopusCustomHeaders.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Octopus.Client.Extensions;
+
+namespace Octopus.Client.Model
+{
+    internal class OctopusCustomHeaders
+    {
+        internal static readonly string EnvVar_TeamCity = "TEAMCITY_VERSION";
```

Comment:
Consider a hashtable / dictionary here, so you can just look through that in `DetermineBuildEvironment` instead of an if for every case

Reply:
👍